test/mixed.js

```diff
@@ -483,6 +483,64 @@ describe('Mixed Types ', () => {
     called.should.equal(true);
   });
 
+  it('tests should be able to access nested parent', async () => {
+    let calledFirst = false;
+    let calledThird = false;
+
+    let testFixture = {
+      firstField: 'test',
+      second: [
+        {
+          thirdField: 'test3',
+        },
+        {
+          thirdField: 'test4',
+        },
+      ],
+    };
+
+    let third = object({
+      thirdField: mixed().test({
+        test() {
+          calledThird = true;
+
+          // I want to do something like this,
+          // this.resolve(ref(this.path),this.from[1].value);
+          // OR probably preferably
+          // this.from[1].resolve(ref(this.path))
+          // but I can't get it working :( ...help?
```

Comment:
what are you trying to test with that broadly? I'd think you want handle ref resolving differently on top of `from` existing. e.g. `ref('../../firstField')`

Reply:
I could only determine if the `from[0].value` is correct when dealing with an array, if 

1. I could resolve the ref or
2. I added the index option [here](https://github.com/jquense/yup/pull/556/files/fa036aa8981567d37ac647bd71b8e5a7573f85e5#diff-dfa1d8cd44cb64e7aff89d961845327dR92)

I didn't particularly like the index option, so I was trying to resolve the ref. If you don't mind leaving that index option in the test context when dealing with an array, then you're right, we can figure out how to resolve the relative ref later.

Reply:
You know what you are when dealing with a parent object, because you know you're own key, but you don't know what you are when dealing with a parent array - hence the idx. Take a look at [this line](https://github.com/jquense/yup/pull/556/files/fa036aa8981567d37ac647bd71b8e5a7573f85e5#diff-a39772825a6753e21b910a44bde28402R514).

Sorry for being AWOL for a while, I'm still a big fan of adding this feature to yup, just want to make sure it's done right.

Reply:
@jquense if you are fine with keeping the index option, then I can remove this comment from the test code and resolve this conversation?

Reply:
yup seems ok I think